File: src/js/is_prime.js
function is_prime(n)
{
    if((n%2==0 && n!=2) || n==1)
        return false;

    for(let i = 3; i*i <= n; i+=2)
    {
        if(n%i==0)
            return false;
    }

    return true;
}

File: src/rust/is_prime.rs
fn is_prime(n: u64) -> bool
{
    if(n<=1 || (n!=2 && n%2==0))
    {
        return false;
    }
    
    for i in (3..n).filter(|x| x%2==1 && x*x <= n)
    {
        if n%i==0
        {
            return false;
        }
    }
    
    return true;
}